### PR TITLE
Fix setup.py, added find_packages again but this time using the exclude= option

### PR DESCRIPTION
**Pull request by AndrewAmmerlaan**
Hi, 

I have been checking out the next branch to prepare an ebuild for the 2.0 release, and to check if all the dependencies are available in the Gentoo repo's. In doing so I ran into a small problem. Apparently, the `packages=` variable in `setup.py` does not include subdirectories. This means that the current `setup.py` file does not install vidify.gui, vidify.api and vidify.player. I have made this PR to fix that, I have reintroduced `find_packages` but I have added the `exclude=` option to prevent installation of the `tests` and `dev` directories, to prevent regression of: https://github.com/marioortizmanero/spotify-music-videos/pull/32.

This time I tested it before making the PR :smile: 

Also, might I suggest to use https://github.com/spyder-ide/qtpy instead of importing PySide2 directly. QtPy is a abstraction layer for PyQt(5) and PySide(2), using this instead of explicitly choosing for PySide2 would increase compatibility with distros that only have one of the two. And it would also prevent users having to install both PyQt5 and PySide2. It is my understanding that PySide2 and PyQt5 are almost identical anyway, and it should be possible to just replace all `import PySide.foobar` with `import qtpy.foobar` without breaking anything. I can test this later, and if you want I can make a PR for this. ([edit] Yes, replacing PySide2 with QtPy works just fine: https://github.com/marioortizmanero/spotify-music-videos/pull/35)

### [EDIT] 
Happy New Year

### [EDIT2] 
I also added a `MANIFEST.in` file, to make sure that all the files in `vidify/gui/res` are installed as well, otherwise the icons etc will be missing.

### [EDIT3] 
I also get this warning when starting vidify:
```
qt.svg: Cannot open file '/home/andrew/vidify/gui/res/icon.svg', because: No such file or directory
qt.svg: Cannot open file '/home/andrew/vidify/gui/res/icon.svg', because: No such file or directory
```
It seems it is looking for icon files in the current working directory instead of in the directory where it is installed (`/usr/lib/python3.7/site-packages/vidify`). I haven't quite figured out how to fix this though.

Maybe something like this would work:
```
os.chdir(os.path.dirname(sys.argv[0]))
```
But I'm not sure if that is the 'correct' solution, there probably is some convention on doing this.

### [EDIT4] 
Is it possible to get the font-colour to correspond to the current qt theme's font colour?
Screenshot_20191231_204335
(The icons are only shown when running from the `/usr/lib/python3.7/site-packages/vidify` directory)

[edit] Yes, we can get the font to match the theme, with this patch:
```
diff --git a/vidify/gui/components.py b/vidify/gui/components.py
index 7b34d4a..0e11f5c 100644
--- a/vidify/gui/components.py
+++ b/vidify/gui/components.py
@@ -52,7 +52,6 @@ class APICard(QGroupBox):
         text = QLabel(description)
         text.setWordWrap(True)
         text.setFont(Fonts.smalltext)
-        text.setStyleSheet(f"color: {Colors.fg};")
         text.setAlignment(Qt.AlignHCenter)
         self.layout.addWidget(text)
 ```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -8,13 +8,15 @@
...
     version=__version__,
-    packages=['vidify'],
+    packages=find_packages(exclude=('tests*', 'images', 'dev*')),
     description='Watch music videos for the songs playing on your device',
     long_description=open('README.md', 'r').read(),
     long_description_content_type='text/markdown',
     url='https://github.com/marioortizmanero/vidify',
     license='LGPL',
 
+    include_package_data=True,
```
Comment (marioortizmanero) [inline setup.py]: Why did you add this?

Comment (AndrewAmmerlaan) [reply setup.py]: That is required for the files in the MANIFEST.in file to be included, without it the file is ingored:
https://python-packaging.readthedocs.io/en/latest/non-code-files.html

**Comment by AndrewAmmerlaan** [comment]
> Is this actually needed? I thought setuptools already took care of this.

Setuptools by default only installs `.py` files, it can also install other files by specifying `package_data=` somewhere in setup.py. However I tried that method as well, and it turns out that including a whole directory is slightly more complex:
```
from glob import glob

setup(
    #...
    data_files = [
        ('target_directory_1', glob('source_dir/**')),
        ('target_directory_2', glob('nested_source_dir/**/*', recursive=True)),
        # etc...
    ],
    #...
)
```
I figured that using the MANIFEST.in file would be more compact.

**Comment by marioortizmanero** [comment]
> Setuptools by default only installs .py files, it can also install other files by specifying package_data= somewhere in setup.py. However I tried that method as well, and it turns out that including a whole directory is slightly more complex:

I mean, if the `MANIFEST.in` file makes it work on more package managers other than pip, like gentoo or arch repos, then let's use it. But if it's only for pip, I'd go with the `setup.py` solution.

> A dark mode is a good idea, however previous experience taught me that when working with colours it is an all or nothing situation: Either specify all colours or none, otherwise you'll always run into trouble on systems that use a non-standard theme. On windows it probably doesn't matter that much because I don't think you can specify a global theme on windows. So maybe this could also be fixed by appending something like if os.name == 'nt': to that line.

I have a module with all the colors and such in `vidify.gui` (the `__init__.py` file), so I'd just modify that to modify its values if dark mode was enabled. It doesn't matter much right now anyway.

**Comment by AndrewAmmerlaan** [comment]
> I mean, if the MANIFEST.in file makes it work on more package managers other than pip, like gentoo or arch repos, then let's use it. But if it's only for pip, I'd go with the setup.py solution.

I have no idea if package managers use the MANIFEST.in file or not, I know gentoo doesn't. Gentoo's ebuilds are basically just a script to run make, setup or whatever, so under the hood it's just executing the setup.py file. I don't know exactly what pip does, but I suppose it also executes the setup.py file under the hood. The line `include_package_data=True,` in setup.py makes setuptools read the MANIFEST.in file. 

I'll see if I can accomplish the same using just setup.py.

**Comment by AndrewAmmerlaan** [comment]
> I'll see if I can accomplish the same using just setup.py.

Fixed

**Comment by marioortizmanero** [comment]
Thank you very much then. I think I can merge & close then. About the relative path to the icons and such, I think I'll add something like this to the `vidify.gui` package:

```python
import os

CURPATH = os.path.dirname(os.path.abspath(__file__))
def abspath(f):
    return os.path.join(CURPATH, f)

class Res:
    icon = abspath("vidify/gui/res/icon.svg")

# prints /home/mario/Downloads/pytest/vidify/gui/res/icon.svg
print(Res.icon)
```

**Comment by AndrewAmmerlaan** [comment]
> Thank you very much. I think I can merge & close then. About the relative path to the icons and such, I think I'll add something like this to the vidify.gui package:

No problem and thank you too :)

One other small thing, it's not really important now, but I thought I'd just dump it here.

I have made the ebuild to also create a `.desktop` file for vidify. As icon it's using the logo files in the images directory. However, logo.png and logo.svg aren't square images. (Almost) All of the other icon files in `/usr/share/icons` and `/usr/share/pixmaps` are square, this seems to be the convention. KDE (and maybe other DE's too) rescale all icons to fit inside the menu's and launchers. This makes the vidify icon be kind stretched:
Screenshot_20200101_145848

It's not really important now, but maybe for in the future it would be nice to have an additional square icon file for .desktop files and launchers. I think it could be just the same logo.{png,svg} with some additional empty space above and below to make it 512x512.

**Comment by marioortizmanero** [comment]
Thanks a lot for pointing this out too! I opened an issue for this in #38 to keep track of the progress. Don't hesitate on opening new issues and pull requests for any other things you notice :)
